[SIE 14] Create Branding component

fernandowinfield opened the pull request:
- Created a Branding component that can be displayed at the top of pages.

- The Branding component includes a logo and a name of the brand. For now I have used the polygon from Sketch as the placeholder for the logo and the name of the brand is passed as a prop to the component. The Branding altogether is a clickable <a> element that navigates to home (`/`).

- Jira task: https://cs6510.atlassian.net/browse/SIE-14

fernandowinfield commented on src/stories/components/Branding/styles.scss:
```diff
@@ -0,0 +1,14 @@
+
+.branding {
+  font-size: $font-size-xl;
+  font-weight: bold;
```
I couldn't find a variable for bold font-weight in the presets that Alessia added so I added this here. But if it would be better to factor that out for later reuse I can try to do that.

stenizwahyudi replied:
I think it should be fine if the one you need is not defined in the global variable. If not, it should be a very easy fix.

alecherryy replied:
@fernandowinfield no variable for font weights. You can use `bold` or `700` to achieve that!

fernandowinfield commented on src/stories/components/Branding/styles.scss:
```diff
@@ -0,0 +1,14 @@
+}
+
+.logo {
+  margin-right: 5px;
```
I decided to use 'pixels' here instead of 'rem' for the separation between the logo and the brand name because I thought the branding should remain consistent throughout the website. But if you think 'rem' is more appropriate here I can easily change it using the function provided by Alessia.

stenizwahyudi replied:
Not really sure about this, but I remember we talked about using rem as it is more reliable than px, so I assume it is better to use margin-right: rem(5px);

alecherryy replied:
I think this okay for now.

fernandowinfield commented on src/stories/components/Branding/styles.scss:
```diff
@@ -0,0 +1,14 @@
+  font-family: $header-font-family;
+}
+
+.logo {
```
In terms of the alignment of the logo, right now it is not perfectly aligned vertically inside the anchor element. In other words, this makes the logo appear higher compared to how it shows on the Sketch design. I wonder if we want the logo to be perfectly aligned vertically. If this is the case I might need help from someone who is experienced in little styling details like this one.

stenizwahyudi replied:
I tried to add some styling to the .logo and it fixes your issue. You can try this:
.logo {
   margin-right: rem(5px);
   position: absolute;
   top: rem(3px);
}

alecherryy replied:
@fernandowinfield you can do the following:
```
.branding {
  align-items: center;
  display: inline-flex;
}
```

fernandowinfield commented on src/stories/components/Branding/Branding.js:
```diff
@@ -0,0 +1,50 @@
+            <img src={Logo}
+                alt={`${text} website logo`}
+                className="logo"/>
+            {text}
```
I decided to just use the text inside the anchor tag (<a><a/>) as the brand name. I wonder if it would be better to use a header tag instead (such as <h1><h1/>)...

stenizwahyudi replied:
Not sure if one is better than the other, but if either one is actually better, I'm sure it should be a very easy fix.

alecherryy replied:
You cannot nest block elements (`<h1>`, `<p>`, etc) inside inline elements (like `<a>`, `<span>`, etc). This is okay for now!

stenizwahyudi commented on src/stories/components/Branding/Branding.js:
```diff
@@ -0,0 +1,52 @@
+    /**
+   * Branding's logo URL
+   */
+    logo: PropTypes.string.isRequired,
```
Should logo also be declared as component parameter above? (see line 11-16)

fernandowinfield replied:
Whoops, I forgot to update that part. We can actually get rid of that parameter if we just load a predefined logo directly from the component. Good catch!

stenizwahyudi commented on src/stories/components/Branding/Branding.stories.js:
```diff
@@ -0,0 +1,20 @@
+export const Default = Template.bind({});
+Default.args = {
+    modifierClasses: '',
+    url: 'src/images/Polygon.png',
```
Maybe change url to logo to be consistent.

stenizwahyudi reviewed (changes requested):
I finished reviewing your pull request for now and suggested some changes. However, I highly recommend to ask for second opinion before actually making any changes.

alecherryy commented on src/stories/components/Branding/Branding.js:
```diff
@@ -0,0 +1,52 @@
+        >
+            <img src={Logo}
+                alt={`${text} website logo`}
+                className="logo"/>
```
@fernandowinfield `className` should be `branding__logo`, following the BEM naming convention.

alecherryy commented on src/stories/components/Branding/Branding.js:
```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+// TODO: replace the import below with the actual logo once it's chosen
+import Logo from '../../../images/Polygon.png';
```
@fernandowinfield let's use an SVG file instead of PNG, that way it won't lose quality. You can export it as an SVG directly from Sketch.

fernandowinfield commented:
- Removed unnecessary component parameter (`logo`)
- Replaced .png logo with a .svg
- Fixed className of logo to follow BEM convention
- Added recommended styling to improve the vertical alignment of the branding elements (logo and name)
